spec/controllers/users_controller_spec.rb

```diff
@@ -120,34 +120,78 @@
           put :update, params: { id: user.to_param, user: valid_attributes }
           expect(response).to redirect_to(user)
         end
-      end
 
-      describe "with invalid params" do
-        it "assigns the user as @user" do
-          allow_any_instance_of(User).to receive(:save).and_return(false)
-          put :update, params: { id: user.to_param, user: { name: 'invalid value' } }
-          expect(assigns(:user)).to eq(user)
-        end
+        context "communication opt-in" do
+          before do
+            Timecop.freeze(Time.new(2002, 10, 31))
```

Comment:
Personal preference, but I'd suggest to just save the `Time.now(etc)` thing into a variable and reuse it later. Just to make sure you're actually talking about the same thing:
```ruby
let(:now) { Time.now(2002, 10, 31) }

before { Timecop.freeze(now) }

after { Timecop.return)

# some thing

it 'some spec' do
  expect(user.attribute).to eq(now)
end
```